### ISSUE TITLE
2.0.5 transpile exception with code transpiling fine with 1.3.1 

### ISSUE THREAD

**Issue by mutex1980** [google/j2objc#934]
First of all thanks for providing such a great tool. I've been working with 1.3.1 and it transpiles my code fine. I tried to switch over to 2.05 and it crashed while transpiling with the stack trace below.

`Exception in thread "main" com.sun.tools.javac.util.Abort
	at com.sun.tools.javac.comp.Check.completionError(Check.java:287)
	at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:584)
	at com.sun.tools.javac.comp.Attr.attribType(Attr.java:638)
	at com.sun.tools.javac.comp.Attr.attribType(Attr.java:631)
	at com.sun.tools.javac.comp.MemberEnter.attribImportType(MemberEnter.java:834)
	at com.sun.tools.javac.comp.MemberEnter.visitImport(MemberEnter.java:558)
	at com.sun.tools.javac.tree.JCTree$JCImport.accept(JCTree.java:571)
	at com.sun.tools.javac.comp.MemberEnter.memberEnter(MemberEnter.java:437)
	at com.sun.tools.javac.comp.MemberEnter.memberEnter(MemberEnter.java:449)
	at com.sun.tools.javac.comp.MemberEnter.visitTopLevel(MemberEnter.java:528)
	at com.sun.tools.javac.tree.JCTree$JCCompilationUnit.accept(JCTree.java:518)
	at com.sun.tools.javac.comp.MemberEnter.memberEnter(MemberEnter.java:437)
	at com.sun.tools.javac.comp.MemberEnter.complete(MemberEnter.java:1038)
	at com.sun.tools.javac.code.Symbol.complete(Symbol.java:574)
	at com.sun.tools.javac.code.Symbol$ClassSymbol.complete(Symbol.java:1037)
	at com.sun.tools.javac.comp.Enter.complete(Enter.java:493)
	at com.sun.tools.javac.main.JavaCompiler.complete(JavaCompiler.java:785)
	at com.sun.tools.javac.main.JavaCompiler$1.complete(JavaCompiler.java:312)
	at com.sun.tools.javac.jvm.ClassReader.fillIn(ClassReader.java:2535)
	at com.sun.tools.javac.jvm.ClassReader.complete(ClassReader.java:2442)
	at com.sun.tools.javac.jvm.ClassReader.access$000(ClassReader.java:76)
	at com.sun.tools.javac.jvm.ClassReader$1.complete(ClassReader.java:240)
	at com.sun.tools.javac.code.Symbol.complete(Symbol.java:574)
	at com.sun.tools.javac.code.Symbol$ClassSymbol.complete(Symbol.java:1037)
	at com.sun.tools.javac.jvm.ClassReader.loadClass(ClassReader.java:2623)
	at com.sun.tools.javac.comp.Resolve.loadClass(Resolve.java:1907)
	at com.sun.tools.javac.comp.Resolve.findGlobalType(Resolve.java:2007)
	at com.sun.tools.javac.comp.Resolve.findType(Resolve.java:2081)
	at com.sun.tools.javac.comp.Resolve.findIdent(Resolve.java:2110)
	at com.sun.tools.javac.comp.Resolve.resolveIdent(Resolve.java:2384)
	at com.sun.tools.javac.comp.Attr.visitIdent(Attr.java:3170)
	at com.sun.tools.javac.tree.JCTree$JCIdent.accept(JCTree.java:2011)
	at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:576)
	at com.sun.tools.javac.comp.Attr.attribType(Attr.java:638)
	at com.sun.tools.javac.comp.Attr.attribType(Attr.java:631)
	at com.sun.tools.javac.comp.MemberEnter.signature(MemberEnter.java:390)
	at com.sun.tools.javac.comp.MemberEnter.visitMethodDef(MemberEnter.java:581)
	at com.sun.tools.javac.tree.JCTree$JCMethodDecl.accept(JCTree.java:778)
	at com.sun.tools.javac.comp.MemberEnter.memberEnter(MemberEnter.java:437)
	at com.sun.tools.javac.comp.MemberEnter.memberEnter(MemberEnter.java:449)
	at com.sun.tools.javac.comp.MemberEnter.finishClass(MemberEnter.java:459)
	at com.sun.tools.javac.comp.MemberEnter.finish(MemberEnter.java:1404)
	at com.sun.tools.javac.comp.MemberEnter.complete(MemberEnter.java:1199)
	at com.sun.tools.javac.code.Symbol.complete(Symbol.java:574)
	at com.sun.tools.javac.code.Symbol$ClassSymbol.complete(Symbol.java:1037)
	at com.sun.tools.javac.comp.Enter.complete(Enter.java:493)
	at com.sun.tools.javac.main.JavaCompiler.complete(JavaCompiler.java:785)
	at com.sun.tools.javac.main.JavaCompiler$1.complete(JavaCompiler.java:312)
	at com.sun.tools.javac.jvm.ClassReader.fillIn(ClassReader.java:2535)
	at com.sun.tools.javac.jvm.ClassReader.complete(ClassReader.java:2442)
	at com.sun.tools.javac.jvm.ClassReader.access$000(ClassReader.java:76)
	at com.sun.tools.javac.jvm.ClassReader$1.complete(ClassReader.java:240)
	at com.sun.tools.javac.code.Symbol.complete(Symbol.java:574)
	at com.sun.tools.javac.code.Symbol$ClassSymbol.complete(Symbol.java:1037)
	at com.sun.tools.javac.jvm.ClassReader.loadClass(ClassReader.java:2623)
	at com.sun.tools.javac.comp.Resolve.loadClass(Resolve.java:1907)
	at com.sun.tools.javac.comp.Resolve.findGlobalType(Resolve.java:2007)
	at com.sun.tools.javac.comp.Resolve.findType(Resolve.java:2081)
	at com.sun.tools.javac.comp.Resolve.findIdent(Resolve.java:2110)
	at com.sun.tools.javac.comp.Resolve.resolveIdent(Resolve.java:2384)
	at com.sun.tools.javac.comp.Attr.visitIdent(Attr.java:3170)
	at com.sun.tools.javac.tree.JCTree$JCIdent.accept(JCTree.java:2011)
	at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:576)
	at com.sun.tools.javac.comp.Attr.visitSelect(Attr.java:3250)
	at com.sun.tools.javac.tree.JCTree$JCFieldAccess.accept(JCTree.java:1897)
	at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:576)
	at com.sun.tools.javac.comp.Attr.visitApply(Attr.java:1825)
	at com.sun.tools.javac.tree.JCTree$JCMethodInvocation.accept(JCTree.java:1465)
	at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:576)
	at com.sun.tools.javac.comp.Attr.visitReturn(Attr.java:1686)
	at com.sun.tools.javac.tree.JCTree$JCReturn.accept(JCTree.java:1384)
	at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:576)
	at com.sun.tools.javac.comp.Attr.attribStat(Attr.java:645)
	at com.sun.tools.javac.comp.Attr.attribStats(Attr.java:661)
	at com.sun.tools.javac.comp.Attr.visitBlock(Attr.java:1124)
	at com.sun.tools.javac.tree.JCTree$JCBlock.accept(JCTree.java:909)
	at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:576)
	at com.sun.tools.javac.comp.Attr.attribStat(Attr.java:645)
	at com.sun.tools.javac.comp.Attr.visitMethodDef(Attr.java:1013)
	at com.sun.tools.javac.tree.JCTree$JCMethodDecl.accept(JCTree.java:778)
	at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:576)
	at com.sun.tools.javac.comp.Attr.attribStat(Attr.java:645)
	at com.sun.tools.javac.comp.Attr.attribClassBody(Attr.java:4364)
	at com.sun.tools.javac.comp.Attr.attribClass(Attr.java:4272)
	at com.sun.tools.javac.comp.Attr.attribClass(Attr.java:4201)
	at com.sun.tools.javac.comp.Attr.attrib(Attr.java:4176)
	at com.sun.tools.javac.main.JavaCompiler.attribute(JavaCompiler.java:1248)
	at com.sun.tools.javac.main.JavaCompiler.attribute(JavaCompiler.java:1221)
	at com.sun.tools.javac.api.JavacTaskImpl.analyze(JavacTaskImpl.java:364)
	at com.sun.tools.javac.api.JavacTaskImpl.analyze(JavacTaskImpl.java:344)
	at com.google.devtools.j2objc.javac.JavacParser.parseFiles(JavacParser.java:167)
	at com.google.devtools.j2objc.pipeline.FileProcessor.processBatch(FileProcessor.java:141)
	at com.google.devtools.j2objc.pipeline.FileProcessor.processInputs(FileProcessor.java:70)
	at com.google.devtools.j2objc.pipeline.TranslationProcessor.processInputs(TranslationProcessor.java:78)
	at com.google.devtools.j2objc.J2ObjC.run(J2ObjC.java:127)
	at com.google.devtools.j2objc.J2ObjC.main(J2ObjC.java:177)`

**Comment by kpgalligan** [comment]
@mutex1980 Exception ends in javac. Check java version

```
java -version
```

Current Java 8 version is 1.8.0_162. If your version is significantly older you'll need to upgrade. That was an issue for me when the parser switched.

**Comment by mutex1980** [comment]
Hi Kevin, indeed my JDK was outdated 1.8.0_25, but after updating to 162 I still get the same error.

**Comment by kpgalligan** [comment]
Do you have more info? How you're calling j2objc and all of the log that comes out?

**Comment by mutex1980** [comment]
I try to identify the transpiled source file triggering the exception and from there the java dependency library it belongs to, because @tomball mentioned an outdated dependency library as reason for #838.

J2objc -v gives me a lot of lines like this:

`an 21, 2018 9:02:17 PM com.google.devtools.j2objc.pipeline.GenerationBatch processSourceFile
AM FEINSTEN: processing <path_2_src_file>.java`

and then says:

`Jan 21, 2018 9:02:17 PM com.google.devtools.j2objc.pipeline.FileProcessor processBatch
AM FEINSTEN: Processing batch of size 86
Exception in thread "main" com.sun.tools.javac.util.Abort
	at com.sun.tools.javac.comp.Check.completionError(Check.java:287)
	at com.sun.tools.javac.comp.Attr.attribTree(Attr.java:584)
	at com.sun.tools.javac.comp.Attr.attribType(Attr.java:638)`

To me this looks like the batch is constructed of 86 files and immedately when the batch is started to be processed i get the exception. This does not allow me to identify the source file to be transpiled the exception comes from. Can I somehow disable batch processing?

BTW: I start j2objc with:

`j2objc --swift-friendly -source 8 --static-accessor-methods --build-closure -v -d <dir> -sourcepath <tons of dirs and jars> -classpath <a jar file> <tons of src files 2 be translated>`

**Comment by kpgalligan** [comment]
I don't think you can "disable batch". You're passing in 86 files to process, right? I would try constructing that call with less files. Try 1 file that doesn't have any dependency on the other 85. If you still crash, that means it's probably something in the dependency parsing. Then start removing stuff from sourcepath and the classpath jar.

Somebody from the j2objc can probably tell you more from your stack trace. The first thing I would try to figure out is if it's bombing on parsing a class or parsing source. If it's a class, its something with your jar. If its source, you'll need to start whiting it down. Go binary (remove half, run, repeat).

What's the jar, btw?

**Comment by tomball** [comment]
Please run the following command, to make sure you don't have a javac issue, separate from j2objc:
```
javac -source 8 -v -d <dir> -sourcepath <tons of dirs and jars> -classpath <a jar file> <tons of src files 2 be translated>
```
You can see I just substituted `javac` for `j2objc` and removed the j2objc-specific flags. It's also the flags that j2objc passes to javac. If you reproduce the problem, then it's unrelated to j2objc.

If this works, then perhaps the issue is with the version of Java 8 installed by the system that built the distribution. The j2objc.sh script used to just use the tools.jar from the developers JDK distribution, but there were too many problems with developers having old JDKs installed. Now, it uses the tools.jar from the JDK installed on the machine building the release, and uses [jarjar](https://code.google.com/archive/p/jarjar/) to repackage the javac classes so they don't conflict with other JDKs. So if the latest Java 8 version of builds your project, post what version worked and we can build a new release with it.

**Comment by mutex1980** [comment]
Following Tom's hint I got a javac error:

`error: duplicate class: org.bouncycastle.asn1.cms.Attribute` 

In fact the class is not duplicate. There are two classes with the same name in different packages. This is a known bug in openjdk [(JDK-8076104)](https://bugs.openjdk.java.net/browse/JDK-8076104)  and oracle java ([JDK-8076104](https://bugs.java.com/bugdatabase/view_bug.do?bug_id=8076104)). 

Java 9 should fix it so since the error is still there with j2objc-2.1.1 I guess it is still based on a broken version. 

Workaround: unzip source jars and pass the extracted folder to source path instead of the original jar

What remains for j2objc is the wish to somehow forward javac compiler errors to the terminal instead of seeing just a crash stack trace as printed above.

**Comment by tomball** [comment]
If javac won't build your project on Java 8, but it works with a Java 9 JDK installed, then by all means use Java 9. We can't fix javac, and j2objc 2.1.1 runs on Java 9.
